src/components/legend/style.js

```diff
@@ -84,6 +84,7 @@ module.exports = function style(s, gd) {
           .enter().append('g')
             .classed('legendpoints', true);
     })
+    .each(styleSpatial)
```

Comment:
[Do not address this comment in this PR - I'm just writing down ideas for future development]

I think `legend/style.js` deserves a refactor. Instead of having the styling logic for all the traces that support legends in here, we should implement a new trace module method named e.g. `_module.legendItem`. This would be more inline with how colorbar styling works. `legend/style.js` would then be pretty trivial: a few legend-wide `.attr()` and then a loop over the items for-each calling their corresponding `_module.legendItem` method. Moreover, by setting a `_module.legendItem` method for all the traces that support legend items, we wouldn't need the `'showLegend'` trace module category.